### PR TITLE
add https with certificate verification support

### PR DESCRIPTION
**Pull request by vishalxebia**

### CODE REVIEW

File: sap/tests/test_sap.py [hunk 1]
```diff
@@ -453,6 +453,83 @@ def test_collect_databases(aggregator, instance):
         aggregator.all_metrics_asserted()
 
 
+def test_collect_only_hosts_https(aggregator, https_instance):
+    # TODO this is needed because the topology retains data across tests
+    topology.reset()
+
+    host_control_url = "https://localhost:1129/SAPHostControl"
+    with requests_mock.mock() as m:
+        m.get(host_control_url + "/?wsdl", text=_read_test_file("wsdl/HttpsHostControl.wsdl"))
```
Comment (TizianoPerrucci) [inline sap/tests/test_sap.py:462]: is the `HttpsHostControl.wsdl` different from `HostControl.wsdl` ?

Comment (vishalxebia) [reply sap/tests/test_sap.py:462]: @TizianoPerrucci yes the URL is changed to `https` URL in the file.

File: sap/stackstate_checks/sap/proxy.py [hunk 2]
```diff
@@ -12,10 +12,23 @@ def __init__(self, url, user, password, verify=False, cert=None, keyfile=None):
...
+                # set by Agent runtime for verification, making this flag `False` doesn't override and doesn't verify
+                # the certificate.
+                session.trust_env = False
             session.cert = (cert, keyfile)
         session.auth = HTTPBasicAuth(user, password)
         wsdl_url = "{0}/?wsdl".format(url)
         self.client = Client(wsdl_url, transport=Transport(session=session, timeout=10))
+        sap_type = wsdl_url.split("/")[-2]
+        address = "/".join(wsdl_url.split("/")[:-2])
+        # ServiceProxy for same host location from config as the host location can be different in WSDL response
```
Comment (TizianoPerrucci) [inline sap/stackstate_checks/sap/proxy.py:26]: please provided both examples urls, and log the urls as well

Comment (vishalxebia) [reply sap/stackstate_checks/sap/proxy.py:26]: provided an example for URLs but logging.

File: sap/stackstate_checks/sap/proxy.py [hunk 3]
```diff
@@ -24,6 +24,24 @@ def __init__(self, url, user, password, verify=False, cert=None, keyfile=None):
         sap_type = wsdl_url.split("/")[-2]
         address = "/".join(wsdl_url.split("/")[:-2])
         # ServiceProxy for same host location from config as the host location can be different in WSDL response
+        # As an Example -
+        #
+        # Case 1.) Type - SAPHostControl
+        #
+        #   URL = http://192.168.0.1:1129
+        #
+        #   SOAP Address location in WSDL response is "http://18.92.32.0:1128/SAPHostControl.cgi"
```
Comment (TizianoPerrucci) [inline sap/stackstate_checks/sap/proxy.py:33]: give both examples http and https, make sure the port is correct

File: sap/stackstate_checks/sap/proxy.py [hunk 4]
```diff
@@ -24,6 +24,24 @@ def __init__(self, url, user, password, verify=False, cert=None, keyfile=None):
...
+        #   "http://192.168.0.1:1129/SAPHostControl.cgi"
+
+        # Case 2.) Type - SAPHostAgent
+        #
+        #   URL = http://192.168.0.1:1129
+        #
+        #   SOAP Address location in WSDL response is "http://18.92.32.0:1128"
+        #   then creating a ServiceProxy with the given URL config, it will become
+        #   "http://192.168.0.1:1129"
+
```
Comment (TizianoPerrucci) [inline sap/stackstate_checks/sap/proxy.py:44]: give both examples http and https, make sure the port is correct

**Pull request merged**